src/store/prefs.js

```diff
@@ -0,0 +1,362 @@
+import Vue from 'vue';
+import { STEVE } from '@/config/types';
+import { clone } from '@/utils/object';
+
+const definitions = {};
+
+export const create = function(name, def, opt = {}) {
+  const parseJSON = opt.parseJSON === true;
+  const asCookie = opt.asCookie === true;
+  const asUserPreference = opt.asUserPreference !== false;
+  const options = opt.options;
+
+  definitions[name] = {
+    def,
+    options,
+    parseJSON,
+    asCookie,
+    asUserPreference,
+    mangleRead:  opt.mangleRead, // Alter the value read from the API (to match old Rancher expectations)
+    mangleWrite: opt.mangleWrite, // Alter the value written back to the API (ditto)
+  };
+
+  return name;
+};
+
+export const mapPref = function(name) {
+  return {
+    get() {
+      return this.$store.getters['prefs/get'](name);
+    },
+
+    set(value) {
+      this.$store.dispatch('prefs/set', { key: name, value });
+    },
+  };
+};
+
+// --------------------
+const parseJSON = true; // Shortcut for setting it below
+const asCookie = true; // Store as a cookie so that it's available before auth + on server-side
+
+// Keys must be lowercase and valid dns label (a-z 0-9 -)
+export const CLUSTER = create('cluster', '');
+export const LAST_NAMESPACE = create('last-namespace', '');
+export const NAMESPACE_FILTERS = create('ns', ['all://user'], { parseJSON });
+export const WORKSPACE = create('workspace', '');
+export const EXPANDED_GROUPS = create('open-groups', ['cluster', 'rbac', 'serviceDiscovery', 'storage', 'workload'], { parseJSON });
+export const FAVORITE_TYPES = create('fav-type', [], { parseJSON });
+export const GROUP_RESOURCES = create('group-by', 'namespace');
+export const DIFF = create('diff', 'unified', { options: ['unified', 'split'] });
+export const THEME = create('theme', 'auto', {
```

Comment:
Well if we're pulling in this code I guess we can implement three-way dark mode (always dark, always light, do whatever the system mode is).